### DISCUSSION TITLE
Fetch data into an atom and modify through Store

### DISCUSSION THREAD

**Discussion by miquik** [pmndrs/jotai#2613] [General]
Hi everyone,
Perhaps what I am doing goes against the spirit of Jotai, but I would like to implement something like this: I would like to retrieve data from the server and save it in an atom (thus causing a re-render of the component). 
Subsequently this data can be modified by classes outside of React (and this I can do using the get/set functions of the ‘Store’).

At some point I would like to cause a re-render of the component using the ‘modified’ data directly from the store. How could I do this?

Thank you

**Comment by dai-shi** [comment]
Sounds like it's an imperative approach. So, instead doing things in `read`, doing them in `write` might help.

https://blog.axlight.com/posts/jotai-tips/
> Tip 6: Derived-like atoms

...may or may not help.

**Comment by miquik** [comment]
Thank you very much @dai-shi but in the end I decided to go another way
